algosdk/atomic_transaction_composer.py

```diff
@@ -785,80 +803,78 @@ def execute(
         self.status = AtomicTransactionComposerStatus.COMMITTED
 
         confirmed_round = resp["confirmed-round"]
-
-        tx_results = [
-            cast(Dict[str, Any], client.pending_transaction_info(tx_id))
-            for tx_id in self.tx_ids
-        ]
-
-        method_results = self.parse_response(tx_results)
+        method_results: List[ABIResult] = []
+
+        for method_index, method in self.method_dict.items():
+            tx_id = self.tx_ids[method_index]
+            result: ABIResult = ABIResult(
+                tx_id=tx_id,
+                raw_value=bytes(),
+                return_value=None,
+                decode_error=None,
+                tx_info={},
+                method=method,
+            )
+            try:
+                tx_info = cast(
+                    Dict[str, Any], client.pending_transaction_info(tx_id)
+                )
+                result = self.parse_result(
+                    method, self.tx_ids[method_index], tx_info
+                )
+            except Exception as e:
+                result.decode_error = e
+            method_results.append(result)
 
         return AtomicTransactionResponse(
             confirmed_round=confirmed_round,
             tx_ids=self.tx_ids,
             results=method_results,
         )
 
-    def parse_response(self, txns: List[Dict[str, Any]]) -> List[ABIResult]:
-        method_results = []
-        for i, tx_info in enumerate(txns):
-            tx_id = self.tx_ids[i]
-            raw_value: Optional[bytes] = None
-            return_value = None
-            decode_error = None
-
-            if i not in self.method_dict:
-                continue
```

Comment:
> This also reduces the total number of pending tx info calls we potentially make, as we only need to call them for ABI Method calls in the transaction group.

Doesn't this branch in the previous code prevent us from getting pending txn info from non-method calls?

Reply:
This is for parsing the txn info - in line 789, I think we previously tried getting all tx info results

Reply:
To be a bit clearer, this was from the most recent commit. 

You're right that originally this branch prevented us from getting the pending txn info call in the try block: https://github.com/algorand/py-algorand-sdk/blob/ce14799e3e6a134a14061730a9385cb1ab844b4d/algosdk/atomic_transaction_composer.py#L533